src/Storages/MergeTree/MergeTreeDeduplicationLog.cpp

```
@@ -231,6 +233,12 @@ std::pair<MergeTreePartInfo, bool> MergeTreeDeduplicationLog::addPart(const std:
         return std::make_pair(info, false);
     }
 
+    if (stopped)
+    {
+        LOG_ERROR(&Poco::Logger::get("MergeTreeDeduplicationLog"), "Storage has been shutdown when we add this part.");
+        return {};
```

Can this block be committed after this? Should we throw an exception instead?

---

`UNFINISHED` or `ABORTED`